### PR TITLE
Update workflow to replace pull_request_target

### PR DESCRIPTION
**Pull request by junqiu-lei**
Signed-off-by: Junqiu Lei <junqiu@amazon.com>

### Description
Use `pull_request` and `workflow_run` to replace `pull_request_target` in workflow. the code in PR will be handled via the pull_request trigger with a manually approve so that it is isolated in an unprivileged environment. The following workflow then starts on `workflow_run` where it is granted access to repository secrets to execute cdk tasks.

Ref: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

Example for this change: https://github.com/junqiu-lei/maps/pull/14

 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by martin-gaievski** [comment]
Does it make sense to get review from author of https://github.com/opensearch-project/maps/pull/24?

**Comment by junqiu-lei** [comment]
> 

Yes, I asked him for reviewing.

File: .github/workflows/pr-approve-test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Save PR number
+        env:
+          PR_NUMBER: ${{ github.event.number }}
+        run: |
+          mkdir -p ./pr
+          echo $PR_NUMBER > ./pr/pr_number
+      - uses: actions/upload-artifact@v3
+        with:
+          name: pr_number
+          path: pr/
```
Comment (Shivamdhar) [inline .github/workflows/pr-approve-test.yml:29]: [nit] extra line at the end of the file.

File: .github/workflows/test-cdk-cd.yml [hunk 2]
```diff
@@ -1,24 +1,56 @@
...
-        - 'tiles-generation/**'
+on:
+  workflow_run:
+    workflows: ["Receive PR Test Approve"]
+    types:
+      - completed
 
 jobs:
-  approve:
+  comment:
```
Comment (Shivamdhar) [inline .github/workflows/test-cdk-cd.yml:10]: what does this action imply?

Comment (junqiu-lei) [reply .github/workflows/test-cdk-cd.yml:10]: I moved the approve step to `pr-approve-test.yml` workflow, here the `comment` in `test-cdk-cd.yml` will help to post the GH actions build url on PR comment.

Comment (junqiu-lei) [reply .github/workflows/test-cdk-cd.yml:10]: Here is a example https://github.com/junqiu-lei/maps/pull/14

File: .github/workflows/pr-approve-test.yml (hunk 1 above)
Comment (junqiu-lei) [reply .github/workflows/pr-approve-test.yml]: updated

**Review by Shivamdhar — APPROVED** [review]

**Review by VijayanB — APPROVED** [review]

**Pull request merged**